src/main/java/org/graylog/integrations/aws/resources/AWSResource.java

```diff
@@ -118,4 +124,43 @@ public Response kinesisHealthCheck(@ApiParam(name = "JSON body", required = true
         KinesisHealthCheckResponse response = kinesisService.healthCheck(heathCheckRequest);
         return Response.accepted().entity(response).build();
     }
+
+    /**
+     * Save the new AWS CloudWatch integration.
+     *
+     *  curl 'http://admin:123123123@localhost:9000/api/plugins/org.graylog.integrations/aws/kinesis/save' \
+     *  -v \
+     *  -X POST \
+     *  -H 'X-Requested-By: just-a-test' \
+     *  -H 'Content-Type: application/json' \
+     *  -H 'Accept: application/json' \
+     *  --compressed \
+     *  --data-binary '{
+     *      "aws_access_key": "",
+     *      "aws_secret_key": "",
+     *      "region": "us-east-1",
+     *      "name": "New Flow Logs",
+     *      "description": "Some flow logs.",
+     *      "aws_input_type": "KINESIS_FLOW_LOGS",
+     *      "stream_name": "flow-logs",
+     *      "batch_size": 10000,
+     *      "assume_role_arn": "",
+     *      "global": false,
+     *      "enable_throttling": false
+     * }'
+     */
+    @POST
+    @Timed
+    @Path("/save")
+    @ApiOperation( value = "Save a new Kinesis input" )
+    @RequiresPermissions(RestPermissions.INPUTS_CREATE)
+    @AuditEvent(type = AuditEventTypes.MESSAGE_INPUT_CREATE)
+    public Response create(@ApiParam(name = "JSON body", required = true)
+                           @Valid @NotNull AWSInputCreateRequest saveRequest) throws Exception {
+
+        awsService.saveInput(saveRequest, getCurrentUser());
+
+        // TODO: Identify if this method needs to return a specific response with the id of the new input.
+        return Response.ok().build();
```

Comment:
Our regular input resource only returns the newly created object ID but in other resources we return the newly created entity. I think the latter makes sense here. That way the caller gets access to the new entity without having to do another GET request to get the details.

Reply:
I do not see the value in returning the entire entity in this case. All of the relevant input details were just posted to this endpoint, so the caller still has those available. Returning the ID definitely make sense though. 

If we were to return the input entity, I think we would need to convert it to an InputSummary value object [similarly to as is being done in InputsResource](https://github.com/Graylog2/graylog2-server/blob/a9a1df0ec08f1b82c973a46446adecd7fd6c9732/graylog2-server/src/main/java/org/graylog2/rest/resources/system/inputs/InputsResource.java#L207). I am just wondering if it's really worth duplicating this block of code right now. What do you think? Definitely open to feedback on this.

Reply:
> I do not see the value in returning the entire entity in this case. All of the relevant input details were just posted to this endpoint, so the caller still has those available. Returning the ID definitely make sense though.

This is more about API conventions than solving an issue for this specific use case. In HTTP APIs you usually get the newly created entity back when you create it. That's what we do in most places of our API. (the inputs resource being one exception)

I leave it up to you. :smiley: 

> If we were to return the input entity, I think we would need to convert it to an InputSummary value object [similarly to as is being done in InputsResource](https://github.com/Graylog2/graylog2-server/blob/a9a1df0ec08f1b82c973a46446adecd7fd6c9732/graylog2-server/src/main/java/org/graylog2/rest/resources/system/inputs/InputsResource.java#L207). I am just wondering if it's really worth duplicating this block of code right now. What do you think? Definitely open to feedback on this.

If we return the newly created entity, we have to duplicate this piece of code or update the server code to make it shareable. The code is masking passwords to avoid exposing them via the API.

Reply:
100% understood. Thanks for the details. This makes sense. I agree with making this consistent with our other API calls and with APIs in general. I'll add the code to return the entity and will follow this pattern in the future too.

Reply:
This has been addressed with https://github.com/Graylog2/graylog-plugin-integrations/pull/93/commits/1fc1735b00a2707331cc71542fbc3292f1d1c695 and server PR https://github.com/Graylog2/graylog2-server/pull/6047.